### PR TITLE
Successfully created a cluster-class based MC for CAPD

### LINKED ISSUE #3214: Unable to create a cluster-class based CAPD cluster
**Bug description**
There currently is no way to deploy a clusterclass based CAPD cluster. This is a bug because it is supposed to be supported functionality

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- (X) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- (X) User Experience
- ( ) Developer Experience

**Expected behavior**

**Steps to reproduce the bug**
1. Set the package-based-lcm featureflag to be true
2. Run `tanzu mc create -i docker` 
3. The CAPD cluster creation fails 

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by saji-pivotal**
### What this PR does / why we need it
This PR adds functionality to create a cluster-class based management cluster for the CAPD provider

### Which issue(s) this PR fixes

Fixes #3214

### Describe testing done for PR
1. Built the changes in the PR locally with `make build-install-cli-local`
2. Ran `./pkg/v1/tkg/test/scripts/cc_hack.sh` to allow for creation of cluster class based clusters
3. Ran `CLUSTER_PLAN=devcc tanzu mc create -i docker -v 9`
4. The MC creation was successful
5. Running `kubectl get cc -A` showed the clusterclasses installed

### Release note

```release-note
Added capability to deploy a cluster-class based management cluster for CAPD
```

### CODE REVIEW

File: pkg/v1/tkg/test/tkgctl/docker_cc/docker_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	//	},
+	//})
+	//Expect(err).ToNot(HaveOccurred())
+	//
+	//err = tkgCtlClient.DeleteCluster(tkgctl.DeleteClustersOptions{
+	//	ClusterName: clusterName,
+	//	Namespace:   constants.DefaultNamespace,
+	//	SkipPrompt:  true,
+	//})
+	//Expect(err).To(BeNil())
```
Comment (anujc25) [inline pkg/v1/tkg/test/tkgctl/docker_cc/docker_suite_test.go:194]: Is there a reason to keep this as commented or should we remove this?

File: pkg/v1/tkg/test/tkgctl/docker_cc/docker_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+	//Expect(optOutStatus.CeipStatus).To(Equal("Opt-out"))
+	//
+	//By(fmt.Sprintf("Creating initial workload cluster %q", clusterName))
+	//
+	//defer os.Remove(clusterConfigFile)
+	//err = tkgCtlClient.CreateCluster(tkgctl.CreateClusterOptions{
+	//	ClusterConfigFile: clusterConfigFile,
+	//	Edition:           "tkg",
+	//})
+	//Expect(err).To(BeNil())
```
Comment (anujc25) [inline pkg/v1/tkg/test/tkgctl/docker_cc/docker_suite_test.go:150]: Is there a reason to keep this as commented or should we remove this?

File: pkg/v1/tkg/managementcomponents/helper.go [hunk 3]
```diff
@@ -37,6 +37,8 @@ func GetTKGPackageConfigValuesFileFromUserConfig(managementPackageVersion string
 		tkrRepoImagePath = fmt.Sprintf("%s/%s", tkgBomConfig.ImageConfig.ImageRepository, tkgBomConfig.TKRPackageRepo.AWS)
 	case constants.InfrastructureProviderAzure:
 		tkrRepoImagePath = fmt.Sprintf("%s/%s", tkgBomConfig.ImageConfig.ImageRepository, tkgBomConfig.TKRPackageRepo.Azure)
+	case constants.InfrastructureProviderDocker:
+		tkrRepoImagePath = fmt.Sprintf("%s/%s", tkgBomConfig.ImageConfig.ImageRepository, tkgBomConfig.TKRPackageRepo.VSphereNonparavirt)
```
Comment (anujc25) [inline pkg/v1/tkg/managementcomponents/helper.go:41]: We should add comment and add assumption on why we are using `tkgBomConfig.TKRPackageRepo.VSphereNonparavirt`.

**Review by anujc25 — COMMENTED** [review]
We need to sync the providers with `package` directory as well. https://github.com/vmware-tanzu/tanzu-framework/tree/main/packages/tkg-clusterclass-docker/bundle/config

File: pkg/v1/tkg/test/tkgctl/docker_cc/docker_suite_test.go (hunk 1 above)
Comment (saji-pivotal) [reply pkg/v1/tkg/test/tkgctl/docker_cc/docker_suite_test.go:194]: Good catch. Deleted

File: pkg/v1/tkg/test/tkgctl/docker_cc/docker_suite_test.go (hunk 2 above)
Comment (saji-pivotal) [reply pkg/v1/tkg/test/tkgctl/docker_cc/docker_suite_test.go:150]: Deleted

File: pkg/v1/tkg/managementcomponents/helper.go (hunk 3 above)
Comment (saji-pivotal) [reply pkg/v1/tkg/managementcomponents/helper.go:41]: Done

**Review by anujc25 — APPROVED** [review]
LGTM

**Pull request merged**